### ISSUE TITLE
Automatically scale icons to be consistent size

### ISSUE THREAD

**Issue by quicksketch** [phocean/TopIcons-plus#34]
Hi there, great extension. New Gnome user here and this was among the first extensions I installed (once I got to wondering where my status icons had gone).

I have 3 icons in my status bar:
- [Franz](http://meetfranz.com/)
- [Shutter](http://shutter-project.org/)
- [Nvidia Settings](https://launchpad.net/~graphics-drivers/+archive/ubuntu/ppa)

For whatever reason, only Franz seems to be the right size. Here's what my top bar looks like:

![screenshot from 2016-08-31 21-47-42](https://cloud.githubusercontent.com/assets/100206/18155656/925dce8e-6fc4-11e6-9d10-be054a236b89.png)

As you can see, Shutter's icon is too large, and Nvidia's is comically small. Is there anything that can be done to ensure consistent sizing across icons? These same applications do not have a problem in Ubuntu Unity.

Running latest git checkout, Ubuntu 16.04.

**Comment by phocean** [comment]
Hi,
I don't know yet how, but there is probably something to do. Thanks for this report.

**Comment by phocean** [comment]
I did a quick hack on the dev branch, which seems promising on my environment.
Can you test it:

```
git checkout dev
make install
```

Reload Gnome with alt-f2 + r.
Thank you.

However, I am skeptical about the nvidia icon.

**Comment by phocean** [comment]
Here is how it is now for me with the latest dev commit.

![capture d ecran de 2016-09-10 18-31-29](https://cloud.githubusercontent.com/assets/4280885/18412029/81f86360-7785-11e6-9256-7f90c45699e6.png)

In my opinion this OK, even if the shutter icon is quite small (but it is also with legacy tray). 

Playing with the size setting in the options (from 18px to 16px) gives a more balanced result:

![capture d ecran de 2016-09-10 18-39-16](https://cloud.githubusercontent.com/assets/4280885/18412057/fddb29c2-7785-11e6-9700-d408579597ec.png)

**Comment by quicksketch** [comment]
Hi @phocean! I'll give this a short when I get the chance.

> However, I am skeptical about the nvidia icon.

FYI the problem with Nvidia's icons were that they were not perfectly square to start with. e.g. They were 200x140px. I managed to "fix" this by editing the png images and just adding margins to make them 200x200, and then they displayed fine.

However I think this is great that this can be fixed in TopIcons, I have no idea where I would lodge a complaint against Nvidia's proprietary driver.

**Comment by phocean** [comment]
Yeah, it is what I suspected. Most similar issues I have seen come from bad icons.
I am not even sure if it is possible to do it in a reasonable manner inside a Gnome-extension. Because fixing icons at runtime would require, I guess, image recognition or cropping with libs like opencve or imagemagick...

I don't think an extension would be the place for that, but rather in the Gnome core instead. But even then, would it make sense?

I don't believe that adding hundreds lines of code and processing is worth it, when it should be the responsibility of the application or theme provider to make compliant icons.

So, reporting the issue to Nvidia is certainly the best thing to do. It is the plain old principle of fixing stuff at the source, rather than trying to mess with things around.

That said, I hope the last change I made with margins, and playing with the size, improves the disproportion for your other icons.

On this point too, it is difficult to make it absolutely perfect as the icon design sometimes varies.

I am closing it for now, but don't hesitate to comment here if the issue is still there, and I would reopen it.

;-)
